main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java

```diff
@@ -47,23 +52,39 @@ public void validate(NoticeContainer noticeContainer) {
       for (int i = 1; i < shapeList.size(); ++i) {
         GtfsShape prev = shapeList.get(i - 1);
         GtfsShape curr = shapeList.get(i);
-        if (prev.hasShapeDistTraveled()
-            && curr.hasShapeDistTraveled()
-            && prev.shapeDistTraveled() >= curr.shapeDistTraveled()) {
+        if (!(prev.hasShapeDistTraveled() || curr.hasShapeDistTraveled())) {
+          continue;
+        }
+        if (prev.shapeDistTraveled() > curr.shapeDistTraveled()) {
+          noticeContainer.addValidationNotice(new DecreasingShapeDistanceNotice(prev, curr));
+          continue;
+        }
+        if (prev.shapeDistTraveled() != curr.shapeDistTraveled()) {
+          continue;
+        }
+        if (sameGpsCoordinates(curr, prev)) {
           noticeContainer.addValidationNotice(
-              new DecreasingOrEqualShapeDistanceNotice(
-                  curr.shapeId(),
-                  curr.csvRowNumber(),
-                  curr.shapeDistTraveled(),
-                  curr.shapePtSequence(),
-                  prev.csvRowNumber(),
-                  prev.shapeDistTraveled(),
-                  prev.shapePtSequence()));
+              new EqualShapeDistanceNotice(prev, curr, SeverityLevel.WARNING));
+          continue;
         }
+        noticeContainer.addValidationNotice(
+            new EqualShapeDistanceNotice(prev, curr, SeverityLevel.ERROR));
       }
     }
   }
 
+  /**
+   * Checks if two {@code GtfsShape} have the same GPS coordinates.
+   *
+   * @param shape the first {@code GtfsShape}
+   * @param otherShape the other {@code GtfsShape}
+   * @return true if both {@code GtfsShape} have the same GPS coordinates, false otherwise.
+   */
+  private static boolean sameGpsCoordinates(GtfsShape shape, GtfsShape otherShape) {
+    return shape.shapePtLon() == otherShape.shapePtLon()
+        && shape.shapePtLat() == otherShape.shapePtLat();
```

Comment:
One of the errors we've been seeing a lot is when there are two shape points that are very close to each other (<1 meter apart) and the shape distance was equal. It could've been that the shape distance units in a particular feed were in kilometers and that due to a rounding process the distance was equal and therefore probably an OK value. It would be great if this check here could return true if the coordinates are less than 1 meter apart.

Reply:
One question is why 1 meter, why not 80cm or 1.20m?

Reply:
1 meter is arbitrary honestly. Maybe @barbeau can provide some more educated insights into GPS precision and measurement.